minui/graphics_fbdev.c

```diff
@@ -179,6 +180,20 @@ static gr_surface fbdev_init(minui_backend* backend) {
 }
 
 static gr_surface fbdev_flip(minui_backend* backend __attribute__((unused))) {
+
+#if defined(RECOVERY_BGRA)
+        // In case of BGRA, do some byte swapping
```

Comment:
Would it be enough to just flip the R and B component inside `gr_color()` (and possibly other places that deal with color)? This way, BGRA support would come at the same cost as RGBA support, whereas with this approach, we have to scan through the whole framebuffer to swap bytes.